plane/src/drone/backend_manager.rs

```diff
@@ -84,12 +152,19 @@ impl BackendManager {
         state: BackendState,
         docker: PlaneDocker,
         state_callback: impl Fn(&BackendState) -> Result<(), Box<dyn Error>> + Send + Sync + 'static,
+        metrics_sender: TypedSocketSender<BackendMetricsMessage>,
```

Comment:
the problem with this is that the `TypedSocketSender` is not durable across reconnects.

One alternative here is to pass the `Arc<StateStore>`, and then call `state_store.get_metrics_sender()` to get the sender for each message at send-time.

Reply:
I've used the same metrics_sender as the one in state_store with an arc, does this work?

Reply:
oh wait a sec, it does not.